public/expressions/overlay_alerts.ts

```diff
@@ -0,0 +1,194 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+import { i18n } from '@osd/i18n';
+import {
+  ExpressionFunctionDefinition,
+  ExprVisLayers,
+} from '../../../../src/plugins/expressions/public';
+import { TimeRange, calculateBounds } from '../../../../src/plugins/data/common';
+import { getClient } from '../services';
+import {
+  VisLayers,
+  VisLayerTypes,
+  VisLayerError,
+  PointInTimeEventsVisLayer,
+} from '../../../../src/plugins/vis_augmenter/public';
+
+type Input = ExprVisLayers;
+type Output = Promise<ExprVisLayers>;
+
+interface Arguments {
+  detectorId: string;
+}
+
+const name = 'overlay_alerts';
+
+export type OverlayAlertsExpressionFunctionDefinition = ExpressionFunctionDefinition<
+  'overlay_alerts',
+  Input,
+  Arguments,
+  Output
+>;
+
+const getAlerts = async (
+  monitorId: string,
+  startTime: number,
+  endTime: number
+): Promise<string> => {
+  const params = {
+    size: 1000,
+    sortField: 'start_time',
+    sortDirection: 'asc',
+    monitorIds: [monitorId],
+  };
+
+  const resp = await getClient().get('/api/alerting/alerts', { query: params });
+
+  if (resp.ok) {
+    const { alerts } = resp;
+
+    // added filter for monitor id since there is a bug in the backend for the alerts api
+    alerts
+      .filter((alert) => alert.start_time >= startTime && alert.start_time <= endTime)
+      .filter((alert) => alert.monitorId == monitorId);
+    return alerts;
+  } else {
+    console.log('error getting alerts:', resp);
+  }
+  return '';
+};
+
+const getMonitorName = async (monitorId: string): Promise<string> => {
+  const resp = await getClient().get('/api/alerting/monitors/' + monitorId);
+  if (resp.ok) {
+    return resp.resp.name as string;
+  } else if (resp.resp === '[alerting_exception] Monitor not found.') {
+    return 'RESOURCE_DELETED'
+  } else if (resp.resp.includes('Do not have permissions to resource') || resp.resp.includes('security_exception')) {
+    return 'FETCH_FAILURE'
```

Comment:
Nitpick: Should we have a constant to store an exhaustive list of valid response statuses that can be interpreted by the dashboards plugin?

Reply:
updated to use the enum